### PR TITLE
dnsdist: Implement TLS Session Ticket Keys management for DoH

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

<strike>Still needs regression tests (in progress).</strike>

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
We should make sure that dnsdist still compiles without libssl after these changes, I realize I didn't test that!

File: pdns/doh.hh [hunk 1]
```diff
@@ -48,11 +48,22 @@ struct DOHFrontend
...
   LibsslTLSVersion d_minTLSVersion{LibsslTLSVersion::TLS10};
+#ifdef HAVE_DNS_OVER_HTTPS
+  std::unique_ptr<OpenSSLTLSTicketKeysRing> d_ticketKeys{nullptr};
+#endif
   std::vector<std::pair<std::string, std::string>> d_customResponseHeaders;
   ComboAddress d_local;
 
   uint32_t d_idleTimeout{30};             // HTTP idle timeout in seconds
   std::vector<std::string> d_urls;
+  std::string d_ticketKeyFile;
```
Comment (chbruyand) [inline pdns/doh.hh]: Maybe also wrap these new members in a `#ifdef HAVE_DNS_OVER_HTTPS #endif` block ?

File: pdns/doh.hh [hunk 2]
```diff
@@ -82,6 +93,7 @@ struct DOHFrontend
   HTTPVersionStats d_http1Stats;
   HTTPVersionStats d_http2Stats;
 
+
```
Comment (chbruyand) [inline pdns/doh.hh]: To remove if any other modification in this file.

File: pdns/doh.hh (hunk 1 above)
Comment (rgacogne) [reply pdns/doh.hh]: I don't like having a lot of `#ifdef` block around, so as long as it doesn't prevent compiling when DoH support is disabled I'd rather leave them alone.

**Pull request merged**